clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -51,6 +52,11 @@ private RecordMetadata(TopicPartition topicPartition, long offset, long timestam
         this.topicPartition = topicPartition;
     }
 
+    @Deprecated
+    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset) {
+        this(topicPartition, baseOffset, relativeOffset, Record.NO_TIMESTAMP, -1, -1, -1);
```

Comment:
Does this actually fix compatibility? I don't think we used relative offsets before, so even though it will compile, it won't behave correctly?